packages/client-core/src/user/menus/ProfileMenu.tsx

```diff
@@ -0,0 +1,692 @@
+    }
+  }
+
+  const enableSocial =
+    authState?.value?.apple ||
+    authState?.value?.discord ||
+    authState?.value?.facebook ||
+    authState?.value?.github ||
+    authState?.value?.google ||
+    authState?.value?.linkedin ||
+    authState?.value?.twitter
+
+  const enableConnect = authState?.value?.emailMagicLink || authState?.value?.smsMagicLink
+
+  return (
+    <div className="relative z-50 h-fit max-h-[60vh] w-[50vw] min-w-[720px] max-w-2xl overflow-y-auto rounded-2xl bg-theme-surface-main p-10">
+      <div className="grid w-full grid-cols-2 gap-x-2">
+        <div className="grid grid-cols-3 gap-x-2">
+          <div className="relative col-span-1 h-20 w-20">
+            <AvatarImage size="fill" src={avatarThumbnail} />
+            <button
+              onClick={() => {
+                PopoverState.showPopupover(<AvatarSelectMenu showBackButton={true} previewEnabled={true} />)
+              }}
+              className="absolute -bottom-3 -right-3 flex h-8 w-8 items-center justify-center rounded-full bg-[#DDE1E5] p-2"
+            >
+              <Edit01Lg className="place-items-center text-black" />
+            </button>
+          </div>
+
+          <div className="col-span-2 grid grid-cols-1 gap-y-1">
+            <Text fontSize="xl" fontWeight="semibold">
+              {hasAdminAccess ? t('user:usermenu.profile.youAreAn') : t('user:usermenu.profile.youAreA')}
+              <span className={commonStyles.bold}>{hasAdminAccess ? ' Admin' : isGuest ? ' Guest' : ' User'}</span>.
+            </Text>
+
+            {acceptedTOS && selfUser?.inviteCode.value && (
+              <Text fontSize="sm">
+                {t('user:usermenu.profile.inviteCode')}: {selfUser.inviteCode.value}
+              </Text>
+            )}
+
+            {acceptedTOS && (
+              <button className="w-fit" onClick={() => showUserId.set(!showUserId.value)}>
+                <Text fontSize="sm">
+                  {showUserId.value ? t('user:usermenu.profile.hideUserId') : t('user:usermenu.profile.showUserId')}
+                </Text>
+              </button>
+            )}
+
+            {acceptedTOS && apiKey?.id && (
+              <button onClick={() => showApiKey.set(!showApiKey.value)} className="w-fit">
+                <Text fontSize="sm">
+                  {showApiKey.value ? t('user:usermenu.profile.hideApiKey') : t('user:usermenu.profile.showApiKey')}
+                </Text>
+              </button>
+            )}
+          </div>
+        </div>
+
+        <div className="grid grid-cols-3 gap-x-2">
+          <button
+            className="col-span-1 flex h-20 w-20 items-center justify-center rounded-full bg-[#616161] p-2"
+            onClick={() => {
+              PopoverState.showPopupover(<SettingsMenu />)
+            }}
+          >
+            <CogLg className="h-10 w-10 text-white" />
+          </button>
+
+          {initialized && (
+            <div className="col-span-2 grid grid-cols-1 gap-y-2">
```

Comment:
both buttons should be 32px tall and 136px wide

Reply:
We try to replicate the design by not hardcoding the values, so that it looks natural in all screen sizes. (Although there are some exceptions). So, unless it looks different from the designs visually, I think it's better to use default layout system as much as possible.

136px looks natural in the 1588x1029 screen size (dimensions are from figma). But it looks unnaturally smaller on 2560x1440 screen, and even small in 4k screens.